app/controllers/searches_controller.rb

```diff
@@ -4,23 +4,35 @@ class SearchesController < ApplicationController
   before_action :ensure_saved_search
 
   def show
-    render pdf: "#{@search.id}",
-         title: "#{@search.title}",
-         show_as_html: params[:format] == 'html',
-             template: 'searches/show.html.haml'
+    respond_to do |format|
+      format.pdf  { render pdf_options }
+      format.html { render pdf_options.merge({ show_as_html: true }) }
+      format.csv  { send_data @search.to_csv, type: Mime::CSV,
```

Comment:
Avoid using `{...}` for multi-line blocks.<br>Block body expression is on the same line as the block start.